stylix/mk-target.nix

```diff
@@ -0,0 +1,102 @@
+{
+  name,
+  humanName,
+  autoEnable ? true,
+  extraOptions ? { },
+  configElements ? [ ],
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
+{
+  options.stylix.targets.${name} = lib.attrsets.unionOfDisjoint {
+    enable = config.lib.stylix.mkEnableTarget humanName autoEnable;
+  } extraOptions;
```

Comment:
Would it be simpler to just have two modules, instead of doing a `unionOfDisjoint` update here?

```nix
imports = [
  {
    options.stylix.targets.${name}.enable = config.lib.stylix.mkEnableTarget humanName autoEnable;
  }
  {
    options.stylix.targets.${name} = extraOptions;
  }
]
```

The module system will already panic if you try to declare the same option multiple times in a way that the options can't be merged, so no need for `unionOfDisjoint`'s assertions.